### PR TITLE
Bump up KC version to 1.5.5 and add few logs in validate()

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
- New version of KC

Release notes:
- [BUG_FIX] Starting the connector didnt use proxy parameters while opening a connection using JDBC - SNOW-363874
- Add provider config. (To be set when using confluent hosted kafka) - SNOW-225155
- Support for tombstone records in native snowflake converters. SNOW-298877

### CODE REVIEW

**Review by sfc-gh-zli — APPROVED** [review]

File: test/connect-log4j.properties [hunk 1]
```diff
@@ -41,3 +41,6 @@ log4j.appender.connectAppender.layout.ConversionPattern=${connect.log.pattern}
 
 log4j.logger.org.apache.zookeeper=ERROR
 log4j.logger.org.reflections=ERROR
+
+# enable debug level logging only for SF's KC
+log4j.logger.com.snowflake.kafka.connector=DEBUG
```
Comment (sfc-gh-rramachandran) [inline test/connect-log4j.properties:46]: Should this be in the release notes? Customers will see more and detailed logs, IMO it is worth mentioning

**Review by sfc-gh-rramachandran — APPROVED** [review]
LGTM!

File: test/connect-log4j.properties (hunk 1 above)
Comment (sfc-gh-japatel) [reply test/connect-log4j.properties:46]: Yes, I plan to add this in a miscellaneous debugging tips.

Comment (sfc-gh-japatel) [reply test/connect-log4j.properties:46]: https://snowflakecomputing.atlassian.net/browse/SNOW-368659

**Pull request merged**